scipy/stats/_sensitivity_analysis.py

```diff
@@ -0,0 +1,588 @@
+    Concatenate to get AB. Which means AB is (2*n, d).
+    """
+    n, d = A.shape
+    AB = np.tile(A, (d, 1, 1))
+    i = np.arange(d)
+    AB[i, :, i] = B[:, i].T
+    AB = AB.reshape(-1, d)
+    return AB
+
+
+def saltelli_2010(
+    f_A: np.ndarray, f_B: np.ndarray, f_AB: np.ndarray
+) -> Tuple[np.ndarray, np.ndarray]:
+    r"""Saltelli2010 formulation.
+
+    .. math::
+
+        S_i = \frac{1}{N} \sum_{j=1}^N
+        f(\mathbf{B})_j (f(\mathbf{AB}^{(i)})_j - f(\mathbf{A})_j)
+
+    .. math::
+
+        S_{T_i} = \frac{1}{N} \sum_{j=1}^N
+        (f(\mathbf{A})_j - f(\mathbf{AB}^{(i)})_j)^2
+
+    Parameters
+    ----------
+    f_A, f_B, f_AB : array_like (n, x)
+        Function evaluations ``n`` being the number of evaluations and ``x``
+        a vector output.
+
+    Returns
+    -------
+    s, st : array_like (x, d)
+        First order and total order Sobol' indices.
+
+    References
+    ----------
+    .. [1] Saltelli, A., P. Annoni, I. Azzini, F. Campolongo, M. Ratto, and
+       S. Tarantola. "Variance based sensitivity analysis of model
+       output. Design and estimator for the total sensitivity index."
+       Computer Physics Communications, 181(2):259-270,
+       :doi:`10.1016/j.cpc.2009.09.018`, 2010.
+    """
+    f_AB = f_AB.reshape(-1, *f_A.shape)
+
+    var = np.var(np.vstack([f_A, f_B]), axis=0)
+
+    s = np.mean(f_B * (f_AB - f_A), axis=1) / var
+    st = 0.5 * np.mean((f_A - f_AB) ** 2, axis=1) / var
+
+    return s.T, st.T
+
+
+@dataclass
+class SobolResult:
+    first_order: np.ndarray
+    total_order: np.ndarray
+    _indices_method: Callable
+    _f_A: np.ndarray
+    _f_B: np.ndarray
+    _f_AB: np.ndarray
+    _A: np.ndarray | None = None
+    _B: np.ndarray | None = None
+    _AB: np.ndarray | None = None
+    _bootstrap_result: BootstrapResult = None  # type: ignore[valid-type]
+
+    def bootstrap(
+        self,
+        confidence_level: DecimalNumber = 0.95,
+        n_resamples: IntNumber = 99
+    ) -> BootstrapResult:  # type: ignore[valid-type]
+        """Bootstrap Sobol' indices to provide confidence intervals.
+
+        Parameters
+        ----------
+        confidence_level : float, default: ``0.95``
+            The confidence level of the confidence intervals.
+        n_resamples : int, default: ``99``
+            The number of resamples performed to form the bootstrap
+            distribution of the indices.
+
+        Returns
+        -------
+        res : BootstrapResult
+            Bootstrap result containing the confidence intervals and the
+            bootstrap distribution of the indices.
+
+        """
+        def statistic(idx):
+            f_A_ = self._f_A[idx]
+            f_B_ = self._f_B[idx]
+
+            n, d = self._A.shape
+            f_AB_ = np.empty((d, *f_A_.shape))
+            for i in range(d):
+                f_AB_[i] = self._f_AB[np.array(idx)+i*n]
+            f_AB_ = f_AB_.reshape(-1, f_A_.shape[1])
```

Comment:
IIUC, this is the same logic as in `sample_AB`. The suggestion (also in tupui/scipy#11) is to make that function do both:

- Form `AB` from `A` and `B` or
- Form `f_AB` from `f_A` and `f_B`

depending on which gets passed in. Then you'd just need to call that here.